### PR TITLE
The DoubleEndedList is no longer treated as package with methods

### PR DESCRIPTION
**Pull request by JKRT**
DoubleEndedList is renamed to DoubleEndedListUtil. 

In addition, it is now a package instead of a hybrid uniontype/package

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/BackendDAE.mo [hunk 1]
```diff
@@ -38,7 +38,7 @@ encapsulated package BackendDAE
 import Absyn;
 import AvlSetPath;
 import DAE;
-import DoubleEndedList;
+import DoubleEndedListUtil.{DoubleEndedList};
```
Comment (sjoelund) [inline OMCompiler/Compiler/BackEnd/BackendDAE.mo]: Should probably not use curly bracers here

**Comment by adrpo** [comment]
I would have probably go for DoubleEndedList as package name and DoubleEndedList.List for the uniontype name instead of DoubleEndedListUtil.DoubleEndedList.

**Comment by JKRT** [comment]
> 
> 
> I would have probably go for DoubleEndedList as package name and DoubleEndedList.List for the uniontype name instead of DoubleEndedListUtil.DoubleEndedList.

The reason that I decided not to do that was to avoid name clashes in Julia. DoubleEndedList would need to have the full path at all places where it is used. E.g DoubleEndedList.List everywhere. Another alternative would be to change the name to something different or if possible have the DoubleEndedList uniontype in the global scope similar to Optional

**Comment by adrpo** [comment]
Ok then, just leave it like it is now.

**Comment by sjoelund** [comment]
DoubleEnded.List
DoubleEnded.prepend
?

**Comment by adrpo** [comment]
We could also move it to List, having List.DoubleEnded, but I'm not sure about this.

**Comment by JKRT** [comment]
> 
> 
> DoubleEnded.List
> DoubleEnded.prepend
> ?

I actually thought about the exact same thing 👯‍♂

**Comment by sjoelund** [comment]
For Julia to work you'd need more and smaller modules

**Comment by JKRT** [comment]
Should I go for DoubleEnded.List approach. I like that the most

**Comment by sjoelund** [comment]
I would. It seems short enough.

**Comment by JKRT** [comment]
Adjusted. However, I opted to use MutableList instead of List. Since list with big L can already be used as a keyword

**Pull request merged**